### ISSUE TITLE
Rails model using Entity.represent and as:

### ISSUE THREAD

**Issue by danieldraper** [ruby-grape/grape-entity#193]
Maybe I'm just trying to use `grape-entity` in a way that is not intended, however I have an entity which represents a Rails model.

    class Entity < Grape::Entity
      expose :email_address, as: :email
    end

and in my Grape API I have the following

```
module API
  module V1
    class Users < Grape::API
      include API::V1::Defaults

      resource :users do
        desc 'Create a User'
        params do
          requires :all, using: User::Entity.documentation
        end

        post do
          transformed_params = User::Entity.represent(params)
          user = User.create! transformed_params
          present user, with: User::Entity
        end
      end
    end
  end
end
```
However I get the error `unknown attribute 'email' for User` is there a way to make represent (or is there another method I've missed) that will return the underlying params without the as:

Basically the issue is we have a legacy database in some places we have `:email` and others `:email_address` we want our request and response params to match across the API and would also like to use `grape-entity` instead of having to manually do this for every endpoint.

Obviously new to Grape and Grape Entity so please close this and point me in the correct direction if I've missed something that already exists.

**Comment by dblock** [comment]
Why wouldn't you just pass `params` into `User.create!` directly? It seems like it does what it intends, transform a hash according to the entity mapping specified.

**Comment by danieldraper** [comment]
@dblock I might not have explained the issue clearly, or maybe I am just misunderstanding here.

I have many tables in my database and some have the field `email`, `email_address`, `emailddr` for argument sake.

I would like to be able to use one attribute name across all endpoints for email called `email` by using `params` I would need to match the param name to the database attribute name which would not allow me to achieve the desired consistency.

I was hopeful Grape::Entity would allow me to basically do the inverse take the presented column names and then give me the raw names back.

```
# Database Table
STRING email_address
```

```
# Request
{
  "email": "john@example.com"
}
```
I'm looking for a magic method that I can use that takes the above request and maps the params to the exposed attribute before any alias.
```
class Entity < Grape::Entity
  expose :email_address, as: :email
end
```

`User.create! Entity.magix(params)`

This method would return a hash like { "email_address": "john@example.com" }

```
# Response
{
  "email": "john@example.com"
}
```

Hopefully this makes a little more sense?

**Comment by dblock** [comment]
I see. I am sure [Roar](https://github.com/apotonick/roar) does that as it both parses and presents models and was designed to do that. You should give it a try. I think Grape::Entity is a bit a long ways off that.

**Comment by danieldraper** [comment]
Thanks @dblock I was able to solve the problem using Roar. For anyone else that finds this problem in the future I used the following (hopefully will clean it up in the future)

```
post do
  user = User.new
  user.extend(User::Representer)
  user.from_json request.body.read
  user.save!
  
  present registration, with: User::Representer
end
```
